src/browser-driver-manager.js

```diff
@@ -1,259 +1,172 @@
-      if (chunk.indexOf('\r') !== -1) {
-        process.stdout.clearLine();
-      }
-      if (lastChunk) {
-        process.stdout.cursorTo(lastChunk.length - 1);
-      }
-      process.stdout.write(chunk);
+  // Create a cache directory if it does not exist on the user's home directory
+  // This will be where environment variables will be stored for the tests
+  // since it is a consistent location across different platforms
+  const BDMCacheDir = getBDMCacheDir();
+  try {
+    await fsPromises.mkdir(BDMCacheDir, { recursive: true });
+  } catch {
+    console.log(`${BDMCacheDir} already exists`);
+  }
 
-      lastChunk = chunk;
+  // Should support for other browsers be added, commander should handle this check.
+  // With only one supported browser, this error message is more meaningful than commander's.
+  if (!browser.includes('chrome')) {
+    throw new Error(
+      `The selected browser, ${browser}, could not be installed. Currently, only "chrome" is supported.`
+    );
+  }
 
-      if (chunk.toString().indexOf('browser-driver-manager error') !== -1) {
-        process.exitCode = 1;
-        reject();
-      }
-    });
-    child.on('close', () => {
-      if (returnValue) {
-        // output all logs except the last one
-        for (let i = 0; i < lines.length - 1; i++) {
-          process.stdout.write(lines[i]);
-        }
-        resolve(lines[lines.length - 1].toString());
-      } else {
-        resolve();
-      }
-    });
-  });
-}
+  const platform = detectBrowserPlatform();
+
+  if (!platform) {
+    throw new Error('Unable to detect browser platform');
+  }
+  // This will sync the browser and chromedriver versions
+  let chromeBuildId;
+  try {
+    chromeBuildId = await resolveBuildId(Browser.CHROME, platform, version);
+  } catch (e) {
+    throw new Error(e);
+  }
 
-/**
- * Install ChromeDriver from npm
- * @param {String} version - Version to install
- * @param {Boolean} verbose - If verbose logging is enabled
- */
-function installChromeDriver(version, verbose) {
-  return new Promise(resolve => {
-    const child = spawn('npm', [
-      'install',
-      '--no-save',
-      `chromedriver@${version}`
-    ]);
-    if (verbose) {
-      child.stdout.on('data', chunk => {
-        process.stdout.write(chunk);
-      });
+  const currentVersion = await getVersion();
+  if (currentVersion) {
+    if (currentVersion === chromeBuildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
     }
-    child.stderr.on('data', chunk => {
-      process.stderr.write(chunk);
-    });
-    child.on('close', () => {
-      resolve();
-    });
+    console.log(
+      `Chrome and Chromedriver versions ${currentVersion} are currently installed. Overwriting.`
+    );
+  }
+
+  const installedChrome = await installBrowser(
```

Comment:
This looks like it can throw an error if passed a version it doesn't support. Example: `chrome@97`. We should catch the error and throw a better message if the error is 404, indicating that the provided version is not supported. It probably also could fail for other reasons so might need to also report a general error.